### ISSUE TITLE
iOS Nested buttons inside buttons not visible to Screen Reader Accessibility Bug

### ISSUE THREAD

**Issue by HannahCarney** [facebook/react-native#32864]
### Description

This is an example whereby the Switch is not accessible and is fully ignored by the screen reader. This doesn't happen in native apps with the same design, but seems to be a massive bug on iOS react native. This is not an issue on android. Any touchable element within another one has the same issue. Our concerns now are having to redesign our entire applications around this to accomodate the <1% of our users who require sceen readers.

Please can you look into this.

### Version

0.66

### Output of `npx react-native info`

System:
    OS: macOS 11.4
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 125.75 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.17.3 - /usr/local/bin/node
    Yarn: 1.22.11 - /usr/local/bin/yarn
    npm: 6.14.13 - /usr/local/bin/npm
    Watchman: Not Found
  Managers:
    CocoaPods: 1.11.2 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
    Android SDK: Not Found
  IDEs:
    Android Studio: 4.2 AI-202.7660.26.42.7486908
    Xcode: 12.5.1/12E507 - /usr/bin/xcodebuild
  Languages:
    Java: Not Found
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.66.0 => 0.66.0 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

Put any touchable element inside another button.

Turn on screen reader

Notice that Switch element or secondary tappable element is completely ignored.

### Snack, code example, screenshot, or link to a repository

``<TouchableOpacity>
   <Switch>
   </Switch>
</TouchableOpacity>``

_No response_

**Comment by cluth17** [comment]
Still an issue. April 12, 2022.

**Comment by blavalla** [comment]
Unfortunately, iOS doesn't support having accessible elements nested. Did you try setting a role of switch and the accessibilityValue property on the <TouchableOpacity> ?

https://reactnative.dev/docs/next/accessibility#accessibilityvalue
